pkg/apis/core/v1beta1/types_namespacedcloudprofile.go

```diff
@@ -0,0 +1,98 @@
+type NamespacedCloudProfileList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list object metadata.
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Items is the list of CloudProfiles.
+	Items []NamespacedCloudProfile `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// NamespacedCloudProfileSpec is the specification of a NamespacedCloudProfile.
+// It must contain exactly one of its defined keys.
+type NamespacedCloudProfileSpec struct {
+	// CABundle is a certificate bundle which will be installed onto every host machine of shoot cluster targeting this profile.
+	// +optional
+	CABundle *string `json:"caBundle,omitempty" protobuf:"bytes,1,opt,name=caBundle"`
+	// Kubernetes contains constraints regarding allowed values of the 'kubernetes' block in the Shoot specification.
+	// +optional
+	Kubernetes *KubernetesSettings `json:"kubernetes" protobuf:"bytes,2,opt,name=kubernetes"`
+	// MachineImages contains constraints regarding allowed values for machine images in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	MachineImages []MachineImage `json:"machineImages" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,3,opt,name=machineImages"`
+	// MachineTypes contains constraints regarding allowed values for machine types in the 'workers' block in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	MachineTypes []MachineType `json:"machineTypes" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,4,opt,name=machineTypes"`
+	// Regions contains constraints regarding allowed values for regions and zones.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	Regions []Region `json:"regions" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,6,opt,name=regions"`
+	// SeedSelector contains an optional list of labels on `Seed` resources that marks those seeds whose shoots may use this provider profile.
+	// An empty list means that all seeds of the same provider type are supported.
+	// This is useful for environments that are of the same type (like openstack) but may have different "instances"/landscapes.
+	// Optionally a list of possible providers can be added to enable cross-provider scheduling. By default, the provider
+	// type of the seed must match the shoot's provider.
+	// +optional
+	SeedSelector *SeedSelector `json:"seedSelector,omitempty" protobuf:"bytes,7,opt,name=seedSelector"`
+	// VolumeTypes contains constraints regarding allowed values for volume types in the 'workers' block in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	VolumeTypes []VolumeType `json:"volumeTypes,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,9,opt,name=volumeTypes"`
+	// A pointer to the NamespacedCloudProfiles parent CloudProfile.
+	// +optional
```

Comment:
If this is optional, then it should be a pointer

Reply:
The parent field is not optional, the `// +optional` was leftover. Is removed in https://github.com/gardener/gardener/pull/9440/commits/bc2bd226df5058cdde6c446ccabd89b024c97d49.